File: TowerOfHanoe/toh.py
#a counter c to store number of iterations
c = 0 
#define a function for recursively solving toh problem
def toh(n,f,t,o):
	#using the global variable c for counting the number of iterations
	global c
	#stopping condition for recursion
	if n == 1:
		print "move disk 1 from %c to %c" %(f,t)		 
		c = c+1
		return
	#recursive call
	toh(n-1,f,o,t)
	#printing the current disk status
	print "move disk %d from %c to %c"%(n,f,t)
	#incrementing the counter
	c = c+1
	#recursive call
	toh(n-1,o,t,f)
	#end of function

#the program driver
#maybe make it a separate function

#taking n as input from the user
n = int(raw_input("Enter the height of the tower at source: "))
#calling the toh function
toh(n,'A','B','C')
#printing the number of recursions
print "Number of steps = ",c

File: TowerOfHanoe/toh.c
/*
This code was written after the python code in the same folder...
*/
#include <stdio.h>
int count = 0; // to maintain counter for counting number of iterations
void toh(int n, char A,char B, char C){
	if (n == 1)
	{
		printf("move block number 1 from %c to %c\n",A,B);
		count++;
		return;
	}
	toh(n-1,A,C,B);
	printf("move block number %d from %c to %c\n",n,A,B);
	count++;
	toh(n-1,C,B,A);
}

int main(int argc, char const *argv[])
{
	int n;
	printf("Enter the height of the tower\n");
	scanf("%d",&n);
	toh(n,'A','B','C');
	printf("The number of iterations: %d\n",count);
	return 0;
}
